src/index.ts

```diff
@@ -10,16 +7,42 @@ import {
   importPublic,
   isValidPrivate,
   isValidPublic,
-} from 'ethereumjs-util'
-import { scrypt } from 'scrypt-js'
+} from '@ethereumjs/util'
+import { base58check } from '@scure/base'
+import * as aes from 'ethereum-cryptography/aes'
+import { keccak256 } from 'ethereum-cryptography/keccak'
+import { getRandomBytesSync } from 'ethereum-cryptography/random'
+import { pbkdf2Sync } from 'ethereum-cryptography/pbkdf2'
+import { scrypt } from 'ethereum-cryptography/scrypt'
+import { sha256 } from 'ethereum-cryptography/sha256'
 
 export { default as hdkey } from './hdkey'
 export { default as thirdparty } from './thirdparty'
-
-const bs58check = require('bs58check')
-const randomBytes = require('randombytes')
 const uuidv4 = require('uuid').v4
 
+const bs58check = base58check(sha256)
+function randomBytes(num: number) {
+  return Buffer.from(getRandomBytesSync(num))
+}
+interface KDFParamsV1 {
+  N: number
+  P: number
+  R: number
+  DkLen: number
+}
+function scryptV1(password: Buffer, salt: Buffer, kdfparams: KDFParamsV1) {
+  const { N, P, R, DkLen } = kdfparams
```

Comment:
Just to be confirmed: the `DkLen` is written different here (upper-case letters) than in the other methods.

Reply:
yeah